src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,184 @@
+            + "Do note that Date and Time must have the specified format\n";
+
+    private TaskList taskList;
+    private Storage storage;
+    private Parser parser;
+
+    /**
+     * Constructor for Ui class.
+     * @param taskList A TaskList object
+     * @param storage A Storage object
+     * @param parser A Parser object
+     */
+    public Ui(TaskList taskList, Storage storage, Parser parser) {
+        this.taskList = taskList;
+        this.storage = storage;
+        this.parser = parser;
+    }
+
+    /**
+     * Prompts user for input and delegates input to Parser.
+     * For use with CLI.
+     */
+    public void echo() {
+
+        Scanner sc = new Scanner(System.in);
+
+        // read duke.txt file and initialise taskList
+        try {
+            storage.init();
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+
+        while (true) {
+            String input = sc.nextLine();
+            if (input.equals("bye")) {
+                break;
+            } else if (input.equals("list")) {
+                System.out.println(taskList.listOut());
+            } else if (input.contains("find")) {
+                System.out.println(taskList.find(input));
+            } else if (input.contains("done")) {
+                try {
+                    System.out.println(taskList.done(input));
+                    storage.save();
+                } catch (DukeException e) {
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     " + e.getMessage());
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println();
+                }
+            } else if (input.contains("delete")) {
+                try {
+                    System.out.println(taskList.delete(input));
+                    storage.save();
+                } catch (DukeException e) {
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     " + e.getMessage());
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println();
+                }
+            } else {
+
+                try {
+                    Task task;
+                    if (input.contains("todo")) {
+                        task = parser.handleToDo(input);
+                    } else if (input.contains("deadline")) {
+                        task = parser.handleDeadline(input);
+                    } else if (input.contains("event")) {
+                        task = parser.handleEvent(input);
+                    } else {
+                        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");

```

Comment:
Perhaps it would be better to standardise the exception message output by DukeException? So that you don't have to type 'OOPS!!!' repetitively in the code. I noticed this was different from the other error messages.